Mathlib/CategoryTheory/Join/Pseudofunctor.lean

```diff
@@ -0,0 +1,182 @@
+/-
+Copyright (c) 2025 Robin Carlier. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Robin Carlier
+-/
+import Mathlib.CategoryTheory.Join.Basic
+import Mathlib.CategoryTheory.Bicategory.Functor.Pseudofunctor
+
+/-!
+# Pseudofunctoriality of categorical joins
+
+In this file, we promote the join construction to two pseudofunctors
+`Join.pseudofunctorLeft` and `Join.pseudoFunctorRight`, expressing its pseudofunctoriality in
+each variable.
+
+-/
+
+universe v₁ v₂ u₁ u₂
+
+namespace CategoryTheory.Join
+
+open Bicategory
+
+/-- The `PrelaxFunctor` structure underlying `Join.pseudofunctorRight`. -/
+@[simps]
+def prelaxFunctorRight (C : Type u₁) [Category.{v₁} C] :
+    PrelaxFunctor Cat.{v₂, u₂} Cat.{max v₁ v₂, max u₁ u₂} where
+  obj D := Cat.of (C ⋆ D)
+  map F := mapPair (𝟭 C) F
+  map₂ := mapWhiskerLeft _
+  map₂_id {x y} f := by
+    apply natTrans_ext <;> aesop_cat
+  map₂_comp η θ := by
+    apply natTrans_ext <;>
+    ( dsimp
+      rw [← mapWhiskerLeft_comp] )
```

Comment:
It seems this can be inlined in the definition `pseudofunctorRight`. The `map₂_comp` can be proven `by apply mapWhiskerLeft_comp`. Could you add a `mapWhiskerLeft_id` lemma? (Also, for the definition of the pseudofunctor, it would be nice to have separate lemmas for both unitors, similarly as you did for the associator.) And similarly for the other variable.